### DISCUSSION TITLE
shallow routing not working well on same page

### DISCUSSION THREAD

**Discussion by keshavgautam** [vercel/next.js#17831] [Help]
I am on ecommerce  site project . when i applied filter there is white screen for few seconds.  i don't want this behavior . 
**I want some trick to apply filter, when i do apply it should update url and product will load from xhr request** 
i want to build [flipkart.com](https://www.flipkart.com/) like website please help me out

**Comment by jamesmosier** [comment]
Do you have any code you can share?

The [example from the  docs](https://nextjs.org/docs/routing/shallow-routing) should help:

```js
// Current URL is '/'
function Page() {
  const router = useRouter()

  useEffect(() => {
    // Always do navigations after the first render
    router.push('/?counter=10', undefined, { shallow: true })
  }, [])

  useEffect(() => {
    // The counter changed!
  }, [router.query.counter])
}
```

**Reply by keshavgautam** [reply]
just look on this   
`

 

    var Dummy_state = {
      paged: this.result.paged,
      page_size: this.result.page_size,
      activefilters: this.ActiveFilters,
      searchstr: this.result.searchstr,
      selectedid: this.result.selectedid,
      sort: this.result.sort,
    };
     
    var currentPath = "/" + this.AuthData.query.id;

    var new_URL = Fn.Build_URL(currentPath, {
      f: Dummy_state,
      t: [3, 4, "hello"],
    });





    Router.push(new_URL,  undefined, { shallow: true })

  }`

**Reply by keshavgautam** [reply]
**Is there  any method**. once page loaded  there is any way for content that  should loaded with xhr and url get update accordingly  and next js render page.
